src/cogent3/app/composable.py

```diff
@@ -1075,7 +1075,9 @@ def _main(self, arg, *args, **kwargs):
     return result
 
 
-def define_app(klass=None, *, app_type: AppType = GENERIC):
+def define_app(
+    klass=None, *, app_type: AppType = GENERIC, skip_not_completed: bool = True
+):
     """decorator for building callable apps
```

Comment:
I think default value for `skip_not_completed` should be False. Because I think most cases is False,

Reply:
I interpret `skip_not_completed=True` to mean that NotCompleted's will not be processed, which is the most common use case. A developer has to take an action of setting it to False for it to change the existing behaviour.

That said, if you have a suggestion for a different name, I'm happy to hear it!

Reply:
Its name is good, I thought default is to not skip.